list.h

```diff
@@ -0,0 +1,252 @@
+#ifndef _LINUX_LIST_H
+#define _LINUX_LIST_H
```

Comment:
I guess we have a license problem here. Linux's GPLv2... Linux kernel's rbtree and list, bit operations are really useful, dunno how we can tackle with license collison